src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -398,11 +400,25 @@ private void InitNativeProcess()
             // redirecting anything. This is a bit tricky as we always run redirected so
             // we have to see if the redirection is actually being done at the topmost level or not.
 
-            //Calculate if input and output are redirected.
+            // Calculate if input and output are redirected.
             bool redirectOutput;
             bool redirectError;
             bool redirectInput;
 
+            _startPosition = new Host.Coordinates();
+
+            try
+            {
+                _startPosition = this.Command.Context.EngineHostInterface.UI.RawUI.CursorPosition;
+                Host.BufferCell[,] bufferContents = this.Command.Context.EngineHostInterface.UI.RawUI.GetBufferContents(
+                    new Host.Rectangle(_startPosition, _startPosition));
+                s_supportScreenScrape = true;
+            }
+            catch (Exception)
+            {
+                // screen scraping not supported on non-Windows or as job
+            }
```

Comment:
I think this check can still be done only once -- we do this check in `CalculateIORedirection` and only if `_runStandAlone` is true. Here the the code I propose:
```CSharp
private void CalculateIORedirection(out bool redirectOutput, out bool redirectError, out bool redirectInput)
{
    ...

    if (NativeCommandProcessor.IsServerSide)
    {
        redirectInput = true;
        redirectOutput = true;
        redirectError = true;
    }
    else if (Platform.IsWindowsDesktop && IsConsoleApplication(this.Path))
    {
        // On Windows desktops, if the command to run is a console application,
        // then allocate a console if there isn't one attached already...
        ConsoleVisibility.AllocateHiddenConsole();

        if (ConsoleVisibility.AlwaysCaptureApplicationIO)
        {
            redirectOutput = true;
            redirectError = true;
        }
    }
    _runStandAlone = !redirectInput && !redirectOutput && !redirectError;

    // !!!!! NOTE !!!!! -- unchanged until here

    if (_runStandAlone)
    {
        if (s_supportScreenScrape == null)
        {
            // Call GetBufferContents() to check and set s_supportScreenScrape to true or false accordingly.
        }

        // if screen scraping isn't supported, we enable redirection so that the output is still transcribed
        // as redirected output is always transcribed
        if (_isTranscribing && (false == s_supportScreenScrape))
        {
            redirectOutput = true;
            redirectError = true;
            _runStandAlone = false;
        }
    }
    
```
So the check of screen scraping support is done only if we are running the native command as standalone application.

In case of running it in a Job, I think the standard input and error are already redirected.

Reply:
I see.  I'll try that.